galileo/src/layer/data_provider/file_cache.rs

```diff
@@ -65,18 +65,39 @@ impl FileCacheController {
     }
 
     fn get_file_path(&self, url: &str) -> PathBuf {
-        let stripped = if let Some(v) = url.strip_prefix("http://") {
-            v
-        } else if let Some(v) = url.strip_prefix("https://") {
-            v
-        } else {
-            url
-        };
-
-        self.folder_path.join(Path::new(stripped))
+        // strip `http` or `https` from url
+        let path = ["http://", "https://"]
+            .iter()
+            .find_map(|prefix| url.strip_prefix(prefix))
+            .unwrap_or(url)
+            // strip query parameters if any
+            .split('?')
+            .next()
+            .unwrap_or(url);
```

Comment:
Although iterator chains are nice and ideomatic, I would prefer plain if-else logic here. It just takes too much mental effort to understand what the iterators do here.